ipn/backend.go

```diff
@@ -21,31 +21,21 @@ import (
 
 type State int
 
+//go:generate go run github.com/nikolaydubina/go-enum-encoding@v1.6.0 -type=State -string
 const (
-	NoState          State = 0
-	InUseOtherUser   State = 1
-	NeedsLogin       State = 2
-	NeedsMachineAuth State = 3
-	Stopped          State = 4
-	Starting         State = 5
-	Running          State = 6
+	NoState          State = iota // json:"NoState"
```

Comment:
Don't use iota when a number crosses the network between processes or hit disk.

These cross the network (from tailscaled to CLI)

Reply:
hm. if this indeed serialized as number due to default JSON serialization as a number, then this PR does not work.

once path I see is using `String()` call (in cli, and other users that use this tailscale libarry in own projects).

<img width="496" alt="image" src="https://github.com/user-attachments/assets/55b9184b-8079-4a9c-ac5e-7008558e3809">

other path uses direct json marshalling as one of the fields in IPNBusWatcher. 
so this is problematic. since JSON will use Marshal/UnmarshalText which are using `String()` value, not int anymore.

<img width="437" alt="image" src="https://github.com/user-attachments/assets/414a9f8e-391d-4e47-a885-c316abd8035e">

<img width="667" alt="image" src="https://github.com/user-attachments/assets/cd00a783-7b25-422f-897c-abcba5305330">

not sure how to make this work without breaking change with old versions.

Reply:
maybe good idea would be splitting this into two enums:

* one that must be int in API
* one that okay be string in API

then converting between them, and using this tool only for string based enum.